cuda_core/tests/requirements-cu12.txt

```diff
@@ -2,3 +2,4 @@ pytest
 # TODO: remove this hack once cupy has a cp313 build
 cupy-cuda12x; python_version < "3.13"
 nvidia-cuda-runtime-cu12  # headers consumed by CuPy
+torch # For PyTorch example
```

Comment:
Similarly need to handle making sure we install a CUDA 12 compatible build of torch for Windows

Reply:
PyTorch Windows cuda packages can be sus unfortunately but the index does work

Reply:
@kkraus14 would it be OK if we defer actually testing against PyTorch until nightly is up (#294)? Installing PyTorch on a per-PR basis would add a significant cost...